### ISSUE TITLE
vpnc resolvconf issues

### ISSUE THREAD

**Issue by sheard** [dnschneid/crouton#372]
I'm having a problem getting vpnc to work after having a clean install of crouton on a new Samsung Chromebook.  It is pretty much the last problem stopping me from working from home.  I've followed exactly the instructions listed in
https://github.com/dnschneid/crouton/wiki/VPNC

But when I run the vpnc script I get:
/run/resolvconf/interface either does not exist or is not a directory

Any ideas?
# 

All working now.  I wrote a script to create the missing directory if not there when starting vpnc.  Added my dns entries needed for local vpn.  Up and running now....

**Comment by murphycc** [comment]
Steve, would you mind showing me an example of the script you wrote for this? I'm getting same error on my Toshiba cb2.

Thanks, Chris

**Comment by sheard** [comment]
Hi Chris,
    The following has continued to work well for me:
    
    I have a script for starting the vpn client. It is copied below, but
    I have put XXXX where local information is required, such as
    nameserver ip addresses and searchable domains.  If you add your
    account to sudoers then you won't be prompted for a password.  I
    keep my ".vpnc.config" file my root directory and mapped keyboard
    shortcuts to the scripts for quick access.
    
    #!/bin/bash
    # Modified 21/09/13, corrected for local nameservers
    
    function dns {
        NS1="xxx.xx.x.xxx"
        NS2="xxx.x.x.x"
        NS3="xxx.xx.x.x"
        OPTIONS="options single-request timeout:1 attempts:5"
        SEARCH="xx.xx.xx.xx"
        echo "## IN VPN MODE" | sudo tee /var/host/shill/resolv.conf
    > /dev/null
        echo "nameserver ${NS1}"  | sudo tee -a
    /var/host/shill/resolv.conf > /dev/null
        echo "nameserver ${NS2}"  | sudo tee -a
    /var/host/shill/resolv.conf > /dev/null
        echo "nameserver ${NS3}"  | sudo tee -a
    /var/host/shill/resolv.conf > /dev/null
        echo "options ${OPTIONS}" | sudo tee -a
    /var/host/shill/resolv.conf > /dev/null
        echo "search ${SEARCH}"   | sudo tee -a
    /var/host/shill/resolv.conf > /dev/null
        echo "## Changed DNS Config"
    }
    
    function main {
    
        # Start sudo with a nice prompt
        sudo -p "Please enter your sudo password to start the VPN: "
    echo "Successfully received password"
    
        # First check if config exists
        if [[ ! -f ~/.vpnc.conf ]]; then
            echo "~/.vpnc.conf not found."
            exit
        fi
    
        # Then check if vpnc is installed
        if [[ ! -f `sudo which vpnc` ]]; then
            echo "vpnc not installed."
            exit
        fi
    
        echo "## Copied current DNS config to /tmp/resolvtmp.conf"
        cp /var/host/shill/resolv.conf /tmp/resolvtmp.conf;
        # Set VPN DNS config
        dns
        if [[ ! -d /run/resolvconf/interface ]]; then
            sudo mkdir /run/resolvconf
        sudo mkdir /run/resolvconf/interface
        fi
        echo "## Starting vpnc in tap mode"
        sudo vpnc --ifmode tap ~/.vpnc.conf
        sleep 2
    }
    
    main; 
    
    
    
    
    I have another script for closing and finishing off, below:
    #!/bin/bash
    
    echo "## Killing vpnc"
    
    sudo killall -9 vpnc
    
    if [ -f /tmp/resolvtmp.conf ]; then
    
        echo "## Going to copy backed up DNS config to
    /var/host/shill/resolv.conf"
    
        sudo cp /tmp/resolvtmp.conf /var/host/shill/resolv.conf
    
    fi
    
    sleep 2
    
    
    
    I'm running trusty on my Samsung Chromebook.
    
    Best wishes,
    Steve
    
    On 07/01/2015 11:46, Chris wrote:
    
    
      
      Steve, would you mind showing me an example of the script you
        wrote for this? I'm getting same error on my Toshiba cb2.
      Thanks, Chris
      —
        Reply to this email directly or view
          it on GitHub.
      
        
          
          
          
           
        
        
      
    
    
    -- 
Steve Sheard,
Department of Engineering Science,
Oxford University,
Parks Road,
Oxford OX1 3PJ
Phone: +44(0)1865 273929

**Comment by murphycc** [comment]
Thanks a lot, Steve.

In the last few days I did find this script also, which looks like the one you might have based yours on:
https://github.com/RaymiiOrg/df/blob/master/vpnc.sh

which looks similar to yours. I actually got VPN finally working last night when I tried the script at that link, though the error I get (resolvconf: Error: /run/resolvconf/interface either does not exist or is not a directory) still exists. However, it does not prevent VPN from working, and I can access IP addresses over the VPN.

I will compare that one a little more to your script and see if yours might help me get rid of this final error.

For your reference, the issue I started a few days ago is at:
https://github.com/dnschneid/crouton/issues/1320

Thanks again,
Chris
